src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1

```diff
@@ -2076,4 +2400,47 @@ class CommonSVTControls: ADOSVTBase {
 
         return $controlResult;
     }
+
+    [psobject]GetResourceApprovalCheck([string] $name, [string] $resourceGroupName, [string] $resourceType, [string] $resourceId)
+    { 
+            $approvalChecks = $this.ResourceApprovalChecks | Where-Object {($_.ResourceId -eq $($resourceId)) -and ($_.ResourceType -eq $($resourceType))}   
+            if(!$approvalChecks){    
+                $url = "https://dev.azure.com/{0}/{1}/_apis/pipelines/checks/queryconfigurations?`$expand=settings&api-version=6.1-preview.1" -f $this.OrganizationContext.OrganizationName, $resourceGroupName;
+                #using ps invoke web request instead of helper method, as post body (json array) not supported in helper method
+                $rmContext = [ContextHelper]::GetCurrentContext();
+                $user = "";
+                $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $user,$rmContext.AccessToken))) 
+                $body = "[{'name':  '$($name)','id':  '$($resourceId)','type':  '$($resourceType)'}]" 
+                if($resourceType -eq 'repository'){
+                    $projectId = ($this.ResourceContext.ResourceId -split "project/")[-1].Split('/')[0]
+                    $body = "[{'name':  '$($name)','id':  '$($projectId +"."+$resourceId)','type':  'repository'}]"
+                }                                       
+                $response = @(Invoke-RestMethod -Uri $url -Method Post -ContentType "application/json" -Headers @{Authorization=("Basic {0}" -f $base64AuthInfo)} -Body $body)
+                $yamlTemplateControl = @()
+                if([Helpers]::CheckMember($response, "count") -and $response[0].count -gt 0){                                                         
+                    try{
+                        $yamlTemplateControl = @($response.value | Where-Object {$_.PSObject.Properties.Name -contains "settings"})
+                    } catch{
+                        $yamlTemplateControl = @()
+                    }
+                }
+                $svtResourceApprovalCheck = [ResourceApprovalCheck]::new();
+                $svtResourceApprovalCheck.ResourceType = $resourceType;
+                $svtResourceApprovalCheck.ResourceId = $resourceId;
+                $svtResourceApprovalCheck.YAMLTemplateControl = $yamlTemplateControl;
+                $this.ResourceApprovalChecks.add($svtResourceApprovalCheck);  
+            }     
+            
+            $approvalChecks = $this.ResourceApprovalChecks | Where-Object {($_.ResourceId -eq $($resourceId)) -and ($_.ResourceType -eq $($resourceType))} 
+            return $approvalChecks;
+    }
+}
+
+#Class used to create Resource Approval Check list inside resolver
+class ResourceApprovalCheck
+{
+	[string] $ResourceId = "";	
+    [string] $ResourceName = ""; 
+    [string] $ResourceType = "";    
+    [PSObject] $YAMLTemplateControl;        
```

Comment:
We can rename this to ApprovalCheckObj as it will store multiple kinds of checks not just yaml template